apps/src/blockly/customBlocks/googleBlockly/variableBlocks.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Constructs the blocks required by the flyout for the variables category.
+ * @param {WorkspaceSvg} workspace The workspace containing variables.
+ * @returns an array of XML block elements
+ */
+export function flyoutCategory(workspace) {
+  const button = document.createElement('button');
+  button.setAttribute('text', '%{BKY_NEW_VARIABLE}');
+  button.setAttribute('callbackKey', 'CREATE_VARIABLE');
+
+  workspace.registerButtonCallback('CREATE_VARIABLE', function (button) {
+    Blockly.Variables.createVariableButtonHandler(button.getTargetWorkspace());
+  });
+
+  const levelConfigBlocks = Array.from(
+    Blockly.cdoUtils.getCustomCategoryBlocksForFlyout('VARIABLE')
+  );
+
+  let mathChangeBlock = null;
+
+  // Find and get the "math_change" block if it exists, and remove it from levelConfigBlocks
+  const updatedLevelConfigBlocks = levelConfigBlocks.filter(block => {
+    if (block.getAttribute('type') === 'math_change') {
+      mathChangeBlock = block;
+      return false; // Exclude the math_change block
+    }
+    return true; // Include other blocks
+  });
+
+  // Call flyoutCategoryBlocks with the mathChangeBlock if found
+  const categoryBlocks = flyoutCategoryBlocks(workspace, mathChangeBlock);
+
+  return [button, ...categoryBlocks, ...updatedLevelConfigBlocks];
+}
+
+/**
+ * Construct the blocks required by the flyout for the variable category.
+ * Modifed version of Blockly.Variables.flyoutCategoryBlocks that skips 'math_change' blocks.
+ * @param workspace The workspace containing variables.
+ * @returns Array of XML block elements.
```

Comment:
Honestly, the main reason was that I was modifying `Blockly.Variables.flyoutCategoryBlocks` from core Blockly which it is still written with XML. https://github.com/google/blockly/blob/master/core/variables.ts#L108C1-L156

I also tried converting the blocks to JSON at one point and hit some issue where there was a conflict of variable IDs. This just worked so I was happy with that. Willing to take another look though!

Reply:
Makes sense. I'm torn between "it works" and "we'll have to rewrite this in the near future". Is my understanding correct that once we switch levelbuilder to json we will need to redo all of this logic? If we can keep this after moving levelbuilder to json it could be ok, otherwise I think we should try to do it in json now, if possible.

Reply:
We'll need to support both, since the toolboxes sources will eventually be either XML or JSON. We'll likely never reach a point where SL levels are all defined in JSON.

Reply:
Done! We now convert to JSON and also define the button and other flyout blocks using JSON. PTAL @molly-moen :)